railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -1261,24 +1261,66 @@ class TwoMigration < ActiveRecord::Migration::Current
         YAML
 
         Dir.chdir(app_path) do
-          animals_db_exists = lambda { rails("runner", "puts !!(AnimalsBase.lease_connection rescue false)").strip }
+          generate_models_for_animals
+
+          assert_not File.exist?("storage/development_animals.sqlite3")
+          assert_not File.exist?("db/animals_schema.rb")
+
+          error = assert_raises do
+            rails "db:migrate:animals" ### Task not defined
+          end
+          assert_includes error.message, "Unrecognized command"
 
+          rails "db:migrate"
+          assert File.exist?("storage/default.sqlite3")
+          assert_not File.exist?("storage/development_animals.sqlite3")
+          assert File.exist?("db/schema.rb")
+          assert_not File.exist?("db/animals_schema.rb")
+
+          rails "db:drop"
+          assert_not File.exist?("storage/default.sqlite3")
+          assert_not File.exist?("storage/development_animals.sqlite3")
+        end
+      end
+
+      test "when database_tasks is false on primary, then run the database tasks on other dbs" do
```

Comment:
Can you also add a test for when there is no primary, and so the first one becomes primary? I _think_ in that case the bug might not be fixed but I haven't tested. Basically `primary?` will fall through to the first config if it's named something else.

```ruby
    def primary?(name) # :nodoc:
      return true if name == "primary"

      first_config = find_db_config(default_env)
      first_config && name == first_config.name
    end 
```